### ISSUE TITLE
Why we do not have support for popular logging platforms?

### ISSUE THREAD

**Issue by Anindyadeep** [Lightning-AI/lit-gpt#576]
So I came across this in the part of the [documentation](https://github.com/Lightning-AI/lit-gpt/blob/0f71b63ebaea0a3114d3d66b7f1d33e330ef4ef2/tutorials/pretrain_redpajama.md?plain=1#L124C1-L126C53)

```
Last, logging is kept minimal in the script. In order to use a particular logger
please refer to <https://lightning.ai/docs/fabric/stable/api/loggers.html> or
call a logging client library like `wandb` directly.
```

Here, why are we not, providing (may be optional) features for direct integration of popular logging platforms? This can be ether:

1. Local path
2. WandB
3. TensorBoard
4. Neptune, etc. 

We can use this extensively for the fine-tuning and evaluation pipeline. For example, for fine-tuning

```bash
python finetune/lora.py \
 --data_dir data/alpaca_libre/ \
 --checkpoint_dir checkpoints/openlm-research/open_llama_7b/ \
 --precision bf16-true
 --log_to wandb/neptune/local/tensorboard
``` 

Or during the case of evaluation, we can put:

```
python eval/lm_eval_harness.py \
   --checkpoint_dir "checkpoints/openlm-research/open_llama_7b/" \
   --precision "bf16-true" \
   --eval_tasks "[truthfulqa_mc]" \
   --batch_size 4 \
   --save_filepath "results-openllama-7b.json"
   --log_to wandb/neptune/local/tensorboard
```

Can we do this?

**Comment by nkasmanoff** [comment]
I can't speak to this decision, but am guessing it's left up to the user since downloading the specific software introduces unnecessary overhead (i.e. what if I use neptune but not wandb).

However, if you are looking for some example code for wandb logging, I have a fork of this repository which has wandb logging for system settings, train loss, validation loss, and some instructional text here: https://github.com/nkasmanoff/llm-finetuning/blob/main/lit-gpt/finetune/lora.py

This is definitely an incomplete version and I may add more logging as I go, but might be useful to you?

**Comment by Anindyadeep** [comment]
Yeah, after putting the issue, I also got that, this could be a possible reason. LitGPT's implementation has been kept open solely for making it a hackable script and right now more focussed towards the NeurIPS implementation. Hence this makes sense. 

Thanks for the script, this helps @nkasmanoff
